### PR TITLE
fixing #issue872

### LINKED ISSUE #872: Using spring.app.name as servicename
The spring boot applications has the option of specifying the `spring.app.name` property, when thats being defined the f8-m-p can use that as service name instead of using `${project.artifactId}`

### Info

* f-m-p version : 3.2.30
* Maven version (`mvn -v`) : 3.3.9
```

```
* Kubernetes / OpenShift setup and version : 1.4.1/3.4.0

### PR DESCRIPTION
**Pull request by theexplorist**
fixes #872

### CODE REVIEW

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 1]
```diff
@@ -32,20 +32,14 @@
...
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (dev-gaur) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: No wildcard imports!

Comment (dev-gaur) [reply enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: You can disable the wildcard imports in your IDE so save yourself from the hassle next time.

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 2]
```diff
@@ -141,14 +135,34 @@ private void addServices(KubernetesListBuilder builder) {
...
+                Properties properties = SpringBootUtil.getSpringBootApplicationProperties(getContext().getProjectClassLoaders().getCompileClassLoader());
+                return properties.getProperty("spring.application.name");
+            }
+        } catch (Exception ex) {
+            log.error("Error while reading the spring-boot configuration", ex);
+        }
+        return null;
+    }
+
+    private String getServiceName(){
```
Comment (dev-gaur) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: space missing between ( ) and {

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 3]
```diff
@@ -141,14 +135,34 @@ private void addServices(KubernetesListBuilder builder) {
...
+        if(getAppName() != null){
```
Comment (dev-gaur) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: spaces missing .. `if (getAppName( ) != null) {`

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 4]
```diff
@@ -141,14 +135,34 @@ private void addServices(KubernetesListBuilder builder) {
...
+            return getAppName();
+        }else{
```
Comment (dev-gaur) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: `} else {`

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 5]
```diff
@@ -141,14 +135,34 @@ private void addServices(KubernetesListBuilder builder) {
         }
     }
 
+    private String getAppName(){
+        try {
+            if (getContext().getProjectClassLoaders().isClassInCompileClasspath(true)) {
+                Properties properties = SpringBootUtil.getSpringBootApplicationProperties(getContext().getProjectClassLoaders().getCompileClassLoader());
+                return properties.getProperty("spring.application.name");
+            }
+        } catch (Exception ex) {
```
Comment (dev-gaur) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: Can you narrow down to a more specific Exception Class type ? Also, Just `e` suffices as the variable name here.

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 6]
```diff
@@ -141,14 +135,34 @@ private void addServices(KubernetesListBuilder builder) {
         }
     }
 
+    private String getAppName(){
+        try {
+            if (getContext().getProjectClassLoaders().isClassInCompileClasspath(true)) {
+                Properties properties = SpringBootUtil.getSpringBootApplicationProperties(getContext().getProjectClassLoaders().getCompileClassLoader());
+                return properties.getProperty("spring.application.name");
```
Comment (dev-gaur) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: @ro14nd @lordofthejars @rohanKanojia can you provide other property names that we should consider for the name ? 

* spring.app.name
* app.name
* spring.service.name ?

WDYT ?

Comment (rohanKanojia) [reply enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: we should stick to standard spring properties, which is `spring.application.name` in this case. If there are other ways there, then we can add them here as well.

Comment (dev-gaur) [reply enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: yeah, that's what I meant.. 

I think `spring.app.name` should be considered as well

**Comment by rohanKanojia** [comment]
@dev-gaur : Can you please take a look if there is anything pending from your side? Otherwise, can we proceed to merge this PR?

**Comment by theexplorist** [comment]
> @dev-gaur : Can you please take a look if there is anything pending from your side? Otherwise, can we proceed to merge this PR?

I will update the PR with requested changes by today so it can be good to merge.

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java (hunk 6 above)
Comment (dev-gaur) [reply enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: @theexplorist can you add a `getProperty("spring.app.name")` as well? Even in #872 its mentioned `spring.app.name` .

File: enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java [hunk 7]
```diff
@@ -141,14 +143,34 @@ private void addServices(KubernetesListBuilder builder) {
...
+            }
+        } catch (Exception ex) {
+            log.error("Error while reading the spring-boot configuration", ex);
+        }
+        return null;
+    }
+
+    private String getServiceName() {
+        if (getAppName() != null) {
```
Comment (lordofthejars) [inline enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultServiceEnricher.java]: Can you skip calling `getAppName` twice? Do something like:

```
String appName = getAppName();
if (appName!=null) {
return appName;
} else {
....
}
```

After that, I think we are ready to merge.

**Pull request merged**